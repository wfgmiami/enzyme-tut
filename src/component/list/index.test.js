import React from 'react';
import List from '.';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

it('renders a list of to dos based on state.toDos', () => {
  const list = shallow(<List />);

  list.setState({
    toDos: [
      {complete: false, text: 'One'},
      {complete: false, text: 'Two'}
    ]
  });

  expect(toJson(list)).toMatchSnapshot()

});
