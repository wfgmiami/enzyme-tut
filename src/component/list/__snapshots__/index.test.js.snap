// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="list"
>
  <h1
    className="list__header"
  >
    todos
  </h1>
  <NewToDo
    submit={[Function]}
  />
  <ToDo
    complete={false}
    deleteToDo={[Function]}
    key="One"
    text="One"
    toggleCompletion={[Function]}
  />
  <ToDo
    complete={false}
    deleteToDo={[Function]}
    key="Two"
    text="Two"
    toggleCompletion={[Function]}
  />
</div>
`;

exports[`renders a list of to dos based on state.toDos 1`] = `
<div
  className="list"
>
  <h1
    className="list__header"
  >
    todos
  </h1>
  <NewToDo
    submit={[Function]}
  />
  <ToDo
    complete={false}
    deleteToDo={[Function]}
    key="One"
    text="One"
    toggleCompletion={[Function]}
  />
  <ToDo
    complete={false}
    deleteToDo={[Function]}
    key="Two"
    text="Two"
    toggleCompletion={[Function]}
  />
</div>
`;
